File: examples/logic/distance.c
#include <stdio.h>
#include <math.h>

double distance(double x1, double y1, double x2, double y2)
{
    return sqrt( pow(x1 + x2, 2) + pow(y1 + y2, 2) );
}

int main(int argc, char* argv[])
{
    if(argc != 5)
    {
        printf("USAGE: %s X1 Y1 X2 Y2", argv[0]);
        return 1;
    }

    double x1, y1, x2, y2, d;

    sscanf(argv[1], "%lf", &x1);
    sscanf(argv[2], "%lf", &y1);
    sscanf(argv[3], "%lf", &x2);
    sscanf(argv[4], "%lf", &y2);

    d = distance(x1, y1, x2, y2);

    printf("The distance from (%.2lf, %.2lf) to (%.2lf, %.2lf) is %.2lf\n", x1, y1, x2, y2, d);

    return 0;
}

File: examples/logic/distance.py
#!/usr/bin/python3

import sys
import math

def distance(x1, y1, x2, y2):
    return math.sqrt( (x1 + x2)**2 + (y1 + y2)**2 )


if __name__ == "__main__":
    if len(sys.argv) != 5:
        print("USAGE: distance.py X1 Y1 X2 Y2")
        sys.exit(1)

    x1 = float(sys.argv[1])
    y1 = float(sys.argv[2])
    x2 = float(sys.argv[3])
    y2 = float(sys.argv[4])

    d = distance(x1, y1, x2, y2)

    print("The distance from ({:.2f},{:.2f}) to ({:.2f}, {:.2f}) is {:.2f}".format(x1, y1, x2, y2, d))
